ts/input/tex/base/BaseItems.ts

```diff
@@ -774,6 +774,45 @@ export class NotItem extends BaseItem {
   }
 }
 
+export class NonscriptItem extends BaseItem {
+
+  /**
+   * @override
+   */
+  public get kind() {
+    return 'nonscript';
+  }
+
+  /**
+   * @override
+   */
+  public checkItem(item: StackItem): CheckType {
+    if (item.isKind('mml') && item.Size() === 1) {
+      let mml = item.First;
+      //
+      //  Space macros like \, wrap with an mstyle to set scriptlevel=0 (so size is independent of level)
+      //
+      if (mml.isKind('mstyle') && mml.notParent) {
+        mml = NodeUtil.getChildren(NodeUtil.getChildren(mml)[0])[0];
+      }
+      if (mml.isKind('mspace')) {
+        //
+        //  If the space is in an mstyle, wrap it in an mrow so we can test is scriptlevel.
+        //  The mrow will be removed in the post-filter.
+        //
+        if (mml !== item.First) {
+          mml = this.create('node', 'mrow', [item.Pop()]);
+          item.Push(mml);
+        }
+        //
+        //  Save the item for alter post-processing
+        //
+        this.factory.configuration.addNode('nonscript', item.First);
+      }
+    }
```

Comment:
As you can see from my comments above, I find the structure very confusing. 
So what do we want to do?
* Any `space` should be marked as `nonscript`
* If the `space` is wrapped in an `mstyle` we want to give it an extra `mrow` and mark that.
(There is no possibility that it is wrapped in multiple `mstyles`?)

Can you restructure the code somewhat to reflect this, even if it introduces some duplications? And/or update the comments?

Reply:
I'll see what I can do.

Reply:
I think what confused me most here is that they are all called `mml`.